index.js

```diff
@@ -179,7 +179,8 @@ class Transaction {
     if (includeSignature) {
       items = this.raw
     } else {
-      if (this._chainId > 0) {
+      const v = ethUtil.bufferToInt(this.v)
+      if ((v === this._chainId * 2 + 35 || v === this._chainId * 2 + 36) && this._common.gteHardfork('spuriousDragon')) {
```

Comment:
Changes made in 812c99d

I didn't quite use two if statements, probably because of stylistic preference. It should be much clearer. I'm fine to go with two `if` statements explicitly if that is your preference :)